stl/inc/xutility

```diff
@@ -4518,6 +4532,18 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
+template <class _Ty1, class _Ty2, class = void>
+_INLINE_VAR constexpr bool _Can_compare_with_operator_equal = false;
+
+template <class _Ty1, class _Ty2>
+_INLINE_VAR constexpr bool
+    _Can_compare_with_operator_equal<_Ty1, _Ty2, void_t<decltype(_STD declval<_Ty1&>() == _STD declval<_Ty2&>())>> =
+        true;
+
+template <class _Ty1, class _Ty2>
+_INLINE_VAR constexpr bool _Is_pointer_address_comparable =
+    (_Is_pointer_address_convertible<_Ty1,
+         _Ty2> || _Is_pointer_address_convertible<_Ty2, _Ty1>) &&_Can_compare_with_operator_equal<_Ty1*, _Ty2*>;

```

Comment:
I guess if you move the `_Can_compare_with_operator_equal<_Ty1*, _Ty2*>` in front, then the lines would wrap better

As far as I can tell it should not really matter what we test first